tests/regression/method.c

```diff
@@ -257,20 +289,249 @@ static result_t cli_compress(
     return result_data(data);
 }
 
+static int advanced_config(
+    ZSTD_CCtx* cctx,
+    buffer_state_t* state,
+    config_t const* config) {
+    for (size_t p = 0; p < config->param_values.size; ++p) {
```

Comment:
minor:
Reading the function, it assumes `cctx` to be correctly initialized.
An `assert()` would be a good way to document this condition.